bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v1/form/FragmentImpl.java

```diff
@@ -0,0 +1,184 @@
+    }
+
+    protected <T> Map<String, T> getChildrenModels(@Nullable SlingHttpServletRequest request, @NotNull Class<T> modelClass) {
+        Map<String, T> models = new LinkedHashMap<>();
+        List<Resource> filteredChildrenResources = getFilteredChildrenResources(fragmentContainer);
+        SlingHttpServletRequest wrappedSlingHttpServletRequest = new SlingHttpServletRequestWrapper(request) {
+
+            @Override
+            public Object getAttribute(String attrName) {
+                if (FormConstants.RESOURCE_CALLER_PATH.equals(attrName)) {
+                    String resourceCallerPath = (String) super.getAttribute(FormConstants.RESOURCE_CALLER_PATH);
+                    // If the attribute is already defined then we're in a nested situation.
+                    // The code for computing the components id uses the root-most resource path
+                    // (because of how componentContext is handled in HTML rendering, so we return
+                    // that.
+                    return (resourceCallerPath != null) ? resourceCallerPath : resource.getPath();
+                }
+                return super.getAttribute(attrName);
+            }
+        };
+        for (Resource child : filteredChildrenResources) {
+            T model = null;
+            if (request != null) {
+                // todo: if possible set i18n form parent to child here, this would optimize the first form rendering
+                model = modelFactory.getModelFromWrappedRequest(wrappedSlingHttpServletRequest, child, modelClass);
+            } else {
+                model = child.adaptTo(modelClass);
+                if (model instanceof Base && i18n != null) {
+                    ((Base) model).setI18n(i18n);
+                }
+            }
+            if (model != null) {
+                models.put(child.getName(), model);
+            }
+        }
+        return models;
+    }
+
+    @Override
+    @JsonIgnore
+    public List<Resource> getFragmentChildren() {
+        if (filteredChildComponents == null) {
+            filteredChildComponents = getFilteredChildrenResources(fragmentContainer);
+        }
+        return filteredChildComponents;
+    }
+
+    @JsonIgnore
+    public String getFragmentTitle() {
+        String fragmentTitle = "";
+        if (fragmentContainer != null) {
+            Resource jcrContentRes = fragmentContainer.getParent();
+            if (jcrContentRes != null) {
+                ValueMap vm = jcrContentRes.getValueMap();
+                fragmentTitle = vm.get("jcr:title", StringUtils.EMPTY);
+            }
+        }
+        return fragmentTitle;
+    }
+
+    @JsonIgnore
+    public List<String> getTitleListOfChildren() {
+        List<String> titleList = new ArrayList<>();
+        if (filteredChildComponents == null) {
+            filteredChildComponents = getFilteredChildrenResources(fragmentContainer);
+        }
+        for (Resource child : filteredChildComponents) {
```

Comment:
Please add unit test cases for all these functions, ideally all has to be covered